<?xml version="1.0"?>
<robot name="MemristorSmall">
    <ros2_control name="HardwareControl" type="system">
        <hardware>
            <plugin>mep3_hardware::MotionHardwareInterface</plugin>
            <param name="kp_linear">90.0</param>
            <param name="ki_linear">1.0</param>
            <param name="kd_linear">2.0</param>

            <param name="kp_angular">110.0</param>
            <param name="ki_angular">1.0</param>
            <param name="kd_angular">0.0</param>

            <param name="update_rate">100.0</param>
        </hardware>
        <joint name="left_motor">
            <state_interface name="position" />
            <command_interface name="velocity" />
        </joint>
        <joint name="right_motor">
            <state_interface name="position" />
            <command_interface name="velocity" />
        </joint>
    </ros2_control>
    <ros2_control name="ax_dynamixels" type="system">
        <hardware>
            <plugin>dynamixel_hardware/DynamixelHardware</plugin>
            <param name="usb_port">/dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A50285BI-if00-port0</param>
            <param name="baud_rate">115200</param>
            <param name="use_dummy">false</param>
            <param name="offset">2.618</param>
        </hardware>
        <joint name="k1">
            <param name="id">69</param>
            <command_interface name="position"/>
            <command_interface name="velocity"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
    </ros2_control>
</robot>
